packages/studio-ui/src/utils/loadUserAssets.tsx

```diff
@@ -37,32 +42,21 @@ export function loadComponents(): Promise<void>[] {
  */
 export async function loadStyling() {
   const studioStore = useStudioStore.getState();
-  const UUIDToFileMetadata = studioStore.fileMetadatas.UUIDToFileMetadata;
-  const pages = studioStore.pages.pages;
-  const layouts = studioStore.layouts.layoutNameToLayoutState;
-
-  for (const fileMetadata of Object.values(UUIDToFileMetadata)) {
-    if (fileMetadata.kind === FileMetadataKind.Component) {
-      await importAndInjectIntoStudio(fileMetadata.cssImports);
-    }
-  }
-
-  for (const page of Object.values(pages)) {
-    await importAndInjectIntoStudio(page.cssImports);
-  }
-
-  for (const layout of Object.values(layouts)) {
-    await importAndInjectIntoStudio(layout.cssImports);
-  }
-}
+  const pages = Object.values(studioStore.pages.pages);
+  const layouts = Object.values(studioStore.layouts.layoutNameToLayoutState);
+  const fileMetadatas = Object.values(
+    studioStore.fileMetadatas.UUIDToFileMetadata
+  );
+  const componentMetadatas = fileMetadatas.filter(
+    (fileMetadata) => fileMetadata.kind === FileMetadataKind.Component
+  ) as ComponentMetadata[];
 
-async function importAndInjectIntoStudio(cssImports: string[]) {
-  for (const filepath of Object.values(cssImports)) {
-    if (
-      document.querySelector(`[${USER_CUSTOM_STYLE_ATTRIBUTE}='${filepath}']`)
-    ) {
-      return;
-    }
+  const userCssFilepaths = getCssImportsFromUserFiles([
+    ...pages,
+    ...layouts,
+    ...componentMetadatas,
+  ]);
+  for (const filepath of userCssFilepaths) {
```

Comment:
I was thinking about this, too.  I guess since we're not addressing the wait for `stylingLoaded` in this PR, we don't need to await on it, and we can just use a stream function. Changed!